rfcs/001-ecosystem-service-credit-module.md

```diff
@@ -0,0 +1,159 @@
+
+### Retire/Burn
+
+The retire or burn operation retires some units of a credit batch that the signer of the transaction holds, making them non-transferrable.
+
+Retiring can be done by either the credit owner, or the credit issuer.
+
+# Rationale
+
+Include an overview of what tradeoffs exist when taking this approach, what benefits come from it, and/or what alternatives were considered.
+
+## Fractional NFT Design
+
+The design described above can be understood as a Fractional NFT (Non-Fungible Token) design where a credit vintage represents a non-fungible asset uniquely identified by its ID and some corresponding metadata. Each credit vintage then consists of a fixed total quantity of credits that are represented as fungible tokens.
+
+If credits were represented as fully fungible tokens, then this would mean all credits of a given credit class would be completely interchangeable with each other regardless of the vintage they came from. This may indeed be the desired behavior in many cases, but representing a credit class in this way would remove the ability to be able to uniquely track credits from a given vintage, which is important for auditing purposes.
+
+The other alternative, to preserve full auditability of all credits, would require a traditional NFT design, where each individual atomic credit unit (e.g. 1 ton of carbon) would be represented by a unique ID as a non-fungible asset. While this would provide the most auditability, it drastically increases the technical complexity, and introduces usability issues (e.g. each individual credit would need to be traded as an individual transaction).
+
+## On-Chain Metadata
+
+This proposal focuses on minimizing on-chain metadata as much as possible. Benefits of this approach include:
+- Reducing technical complexity, thus reducing development time
+- Allowing for specific use cases to drive future functionality
+
+Specifically, the metadata fields in the credit class and credit vintage could be designed to be much more explicit, with several required on-chain metadata fields. Since we are not planning on implementing on-chain scraping of credit metadata, it seems more appropriate to allow for arbitrary bytes, and allow for standards for metadata best practices to evolve over a longer period of time once we can generalize across several use cases.
+
+## Complex State Machines
+
+Another design decision made in this proposal was to only implement two possible states for credits: “tradable” and “retired”. These are the two known states that we need to be able to manage for our first use case with the registry. While it is assumed that there may be more complex state machines that we will need to support at a later date, we chose to start with satisfying the current use case for the same reasons illustrated above (reducing complexity, and allowing future use cases to drive more complex functionality or generalizations).
```

Comment:
```suggestion
Another design decision made in this proposal was to only implement two possible states for credits: “tradable” and “retired”. These are the two known states that we need to be able to manage for our first use case with the registry. While it is assumed there may be more complex states that we will need to support at a later date, we chose to start with satisfying the current use case for the same reasons illustrated above (reducing complexity, and allowing future use cases to drive more complex functionality or generalizations).
```

i think you meant only `states` here instead of `state machines`?

Reply:
No - meant state machines (Right now the state machine is `tradable` -> `retired`, and that's the only allowable state transition possible).  It may be in the future that we have other state transitions possible (to newly defined states).